### PR TITLE
io: Simplify crate docs and add README

### PR DESCRIPTION
**Pull request by tcharding**
Simplify the docs in `lib.rs` and copy them into a minimal README file.

### CODE REVIEW

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/rust-bitcoin/rust-bitcoin/pull/2252/commits/dddf50f97f7592a3254e52d9349074b78026a36e

**Review by Kixunil — DISMISSED** [review]
I believe the README path needs to change.

File: io/Cargo.toml [hunk 1]
```diff
@@ -8,8 +8,9 @@ documentation = "https://docs.rs/bitcoin-io/"
 description = "Simple I/O traits for no-std (and std) environments"
 categories = ["no-std"]
 keywords = [ "io", "no-std" ]
-readme = "README.md"
-edition = "2018"
+readme = "../README.md"
```
[the remark this answers is not in the document]
Comment (tcharding) [reply io/Cargo.toml:11]: Ugh, I did that before writing the readme, face palm.

File: io/Cargo.toml [hunk 2]
```diff
@@ -8,8 +8,9 @@ documentation = "https://docs.rs/bitcoin-io/"
...
+edition = "2021"
+rust-version = "1.56.1"
```
[the remark this answers is not in the document]
Comment (tcharding) [reply io/Cargo.toml:13]: Oh I assumed it got picked up from the workspace. I'll check, if it isn't it should be.

**Comment by tcharding** [comment]
Re the docs, I pure cut'n'pasted these from `lib.rs` without thinking too much about it. I can put in more effort, thanks for the review.

**Comment by Kixunil** [comment]
Ah, OK, I didn't notice lib.rs has the same doc

**Comment by tcharding** [comment]
Force push now changes the crate level rustdoc and copies the new content to `README.md`. Please re-review.

File: io/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Rust-Bitcoin IO Library
+=======================
+
+The `std::io` module is not exposed in `no-std` Rust, building `no-std` applications which require
```
Comment (apoelstra) [inline io/README.md:4]: In b82d54985b4dba33bf1bc312dd1a60ce5a6bb40c:

Should add `so ` before `building`

Comment (apoelstra) [reply io/README.md:4]: and below

**Comment by tcharding** [comment]
Used 'so' as suggested. No other changes.

**Review by apoelstra — APPROVED** [review]
ACK d480adaf25f5e1a43a8c74ffebf6b9ee4c3672d4

**Review by Kixunil — APPROVED** [review]
ACK d480adaf25f5e1a43a8c74ffebf6b9ee4c3672d4

**Pull request merged**